src/evaluators/CallExpression.js

```diff
@@ -121,7 +121,12 @@ function generateRuntimeCall(
     let argStart = 1;
     if (thisArg instanceof Value) {
       if (typeof propName === "string") {
-        callFunc = t.memberExpression(nodes[0], t.identifier(propName), !t.isValidIdentifier(propName));
+        let isValidIdentifier = t.isValidIdentifier(propName);
+        callFunc = t.memberExpression(
```

Comment:
I think this would be clearer if you branched before memberExpression call and turned it into two possible calls.